jenkins/Hono-Website-Pipeline.groovy

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env groovy
+
+/*******************************************************************************
+ * Copyright (c) 2016, 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+/**
+ * Builds Hono's web site using Hugo every night (between 3 and 4 AM) and publishes to Eclipse.
+ *
+ */
+
+node {
+    def utils = evaluate readTrusted("jenkins/Hono-PipelineUtils.groovy")
+    properties([buildDiscarder(logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '', daysToKeepStr: '', numToKeepStr: '3')),
+                pipelineTriggers([cron('TZ=Europe/Berlin \n # every night between 3 and 4 AM \n H 3 * * *')])])
+    try {
+        deleteDir()
+        build()
+        currentBuild.result = 'SUCCESS'
+    } catch (err) {
+        currentBuild.result = 'FAILURE'
+        echo "Error: ${err}"
+    }
+    finally {
+        echo "Build status: ${currentBuild.result}"
+        utils.notifyBuildStatus()
+    }
+}
+
+/**
+ * Build stages for Hono's web site pipeline using Hugo
+ *
+ */
+def build() {
+
+    stage('Checkout to sub-directory "hono"') {
+        echo "Check out master to subdirectory [hono]"
+        checkout([$class                           : 'GitSCM',
+                  branches                         : [[name: 'master']],
+                  doGenerateSubmoduleConfigurations: false,
+                  extensions                       : [[$class: 'RelativeTargetDirectory', relativeTargetDir: 'hono']],
+                  submoduleCfg                     : [],
+                  userRemoteConfigs                : [[credentialsId: 'bebef3c5-da22-425c-8554-f62d0b3a9608',
```

Comment:
where does the credentialsId come from? Is this really something that we want to include in the code instead of e.g. passing it in as a configuration parameter?

Reply:
An internal unique ID by which credentials are identified from jobs and other configuration. Normally left blank, in which case an ID will be generated, which is fine for jobs created using visual forms. Useful to specify explicitly when using credentials from scripted configuration (Reference: CredentialsPlugin).

The credentials Id that I used was already available and I think it was configured by Eclipse. I think its a good idea to parameterize the Id too.